src/include/execution_kernel.hpp

```diff
@@ -568,8 +598,13 @@ __global__ void executionKernel([[maybe_unused]] int rank /*for debug*/, T* inpu
                                op.inputChannelIndexes, op.outputOffsets, op.inputOffsets, op.nOutputs, op.nInputs,
                                op.size, false);
     } else if (op.type == OperationType::PUT_PACKET) {
-      handlePutPacket<PacketType>(scratchSize, smChannels, proxyChannels, op.outputChannelIndexes, op.outputOffsets,
-                                  op.inputOffsets, op.nOutputs, op.size, op.channelType, flag);
+      if (op.inputBufferType == BufferType::SCRATCH && op.channelType == ChannelType::SM) {
```

Comment:
Do we need to introduce new OP to handle this. Current implementation depends on channel type and buffer type, seems not a good choice.